server/mqtt.go

```diff
@@ -243,14 +243,16 @@ type sessPersistRecord struct {
 }
 
 type mqttJSA struct {
-	mu      sync.Mutex
-	id      string
-	c       *client
-	sendq   chan *mqttJSPubMsg
-	rplyr   string
-	replies sync.Map
-	nuid    *nuid.NUID
-	quitCh  chan struct{}
+	mu        sync.Mutex
+	id        string
+	c         *client
+	sendq     chan *mqttJSPubMsg
+	rplyr     string
+	replies   sync.Map
+	nuid      *nuid.NUID
+	quitCh    chan struct{}
+	domain    string // Domain or possibly empty. This is added to session subject.
```

Comment:
Granted it would probably be more work, but I thought we discussed that we would actually store a prefix here (either `$JS.API` or actual `$JS.<domain>.API`) and add this prefix to all API subjects. We would have to replace existing references to the constants such as `JSApiStreamCreateT` to something like `mqttAPIStreamCreateT` that would be defined as the same than JSApiStreamCreateT but without the prefix, in this case `%s.STREAM.CREATE.%s`.  So in place where we currently have:
```
jsa.newRequest(mqttJSAStreamCreate, fmt.Sprintf(JSApiStreamCreateT, cfg.Name), 0, cfgb)
```
we would do:
```
jsa.newRequest(mqttJSAStreamCreate, fmt.Sprintf(mqttAPIStreamCreateT, jsa.apiPfx, cfg.Name), 0, cfgb)
```
or something similar.

I agree that the places impacted currently should be "low volume", but still, that makes use of more sprintf/memory allocs if you trim the prefix and create the actual prefix in each requests (in newRequestEx).

We could go with your approach first and improve later if you think it is best to minimize changes in this PR.

Reply:
I will submit a separate PR once this one is oked.